eftest/src/eftest/runner.clj

```diff
@@ -16,24 +16,44 @@
   (or (-> v meta :eftest/synchronized true?)
       (-> v meta :ns meta :eftest/synchronized true?)))
 
+(defn- known-slow? [v]
+  (or (-> v meta :eftest/slow true?)
+      (-> v meta :ns meta :eftest/slow true?)))
+
 (defn- failed-test? []
   (or (< 0 (:error @test/*report-counters* 0))
       (< 0 (:fail @test/*report-counters* 0))))
 
+(defn- run-test-and-report-time
+  [test-warn-length start-time v]
+  (let [result   (test/test-var v)
+        end-time (System/nanoTime)
+        duration (/ (- end-time start-time) 1e6)]
+    (when (and (not (known-slow? v))
+               (number? test-warn-length)
+               (<= test-warn-length duration))
+      (binding [clojure.test/*testing-vars* [v]]
+        (test/report {:type     :long-test
+                      :duration duration
+                      :var      v})))
+    result))
+
 (defn- test-vars
-  [ns vars {:as opts :keys [fail-fast? capture-output?] :or {capture-output? true}}]
-  (let [once-fixtures (-> ns meta ::test/once-fixtures test/join-fixtures)
+  [ns vars {:as opts :keys [fail-fast? capture-output? test-warn-length] :or {capture-output? true}}]
+  (let [
+        once-fixtures (-> ns meta ::test/once-fixtures test/join-fixtures)
         each-fixtures (-> ns meta ::test/each-fixtures test/join-fixtures)
         report        (synchronize test/report)
         test-var      (fn [v]
                         (when-not (and fail-fast? (failed-test?))
-                          (each-fixtures
-                           (if capture-output?
-                             #(binding [test/report report]
-                                (capture/with-test-buffer
-                                  (test/test-var v)))
-                             #(binding [test/report report]
-                                (test/test-var v))))))]
+                          (let [start-time (System/nanoTime)]
+                            (each-fixtures
+                             (if capture-output?
+                               #(binding [test/report report]
+                                  (capture/with-test-buffer
+                                    (run-test-and-report-time test-warn-length start-time v)))
```

Comment:
I think this would be better handled as a wrapper function around the `test-var` function defined in this let clause.

Reply:
My reasoning for sticking it in here, was so that we weren't timing tests that were being skipped (i.e. re-using the code to skip testing from `(and fail-fast? (failed-test?))`.

Reply:
You could still create a wrapper function and put it after the `fail-fast` test. It's better to have timing logic in one place.

Reply:
Ok, all timing logic is in one function now.